aikau/src/main/resources/alfresco/testing/MockXhr.js

```diff
@@ -73,14 +79,19 @@ define(["dojo/_base/declare",
          this.loadBinaryData();
 
          // Set-up a fake server to handle all the responses...
-         this.server = sinon.fakeServer.create();
-         this.server.autoRespond = true;
-         this.setupServer();
-
+         var server = this.server = sinon.fakeServer.create();
          if (this.respondAfter)
          {
             this.server.autoRespondAfter = this.respondAfter;
          }
+         server.autoRespond = true;
+         server.xhr.useFilters = true;
+
+         // Adds a filter to allow dynamic dependency requests to be processed normally...
+         server.xhr.addFilter(function(method, url) {
+           return !!url.match(/surf\/dojo\/xhr\/dependencies/);
```

Comment:
Purely for academic reasons, you might be interested to know that it'd be twice as fast to use RegExp.exec ...

```javascript
return !!/surf\/dojo\/xhr\/dependencies/.exec(url);
```